rust/agama-server/src/storage/web.rs

```diff
@@ -0,0 +1,51 @@
+//! This module implements the web API for the storage service.
+//!
+//! The module offers two public functions:
+//!
+//! * `storage_service` which returns the Axum service.
+//! * `storage_stream` which offers an stream that emits the storage events coming from D-Bus.
+
+use agama_lib::{error::ServiceError, storage::StorageClient};
+use axum::{extract::State, routing::get, Json, Router};
+
+use crate::{
+    error::Error,
+    web::{
+        common::{issues_router, progress_router, service_status_router, EventStreams},
+        Event,
+    },
+};
+
+pub async fn storage_streams(dbus: zbus::Connection) -> Result<EventStreams, Error> {
+    let result: EventStreams = vec![]; // TODO:
+    Ok(result)
+}
+
+#[derive(Clone)]
+struct StorageState<'a> {
+    client: StorageClient<'a>,
+}
+
+/// Sets up and returns the axum service for the software module.
+pub async fn storage_service(dbus: zbus::Connection) -> Result<Router, ServiceError> {
+    const DBUS_SERVICE: &str = "org.opensuse.Agama.Storage1";
+    const DBUS_PATH: &str = "/org/opensuse/Agama/Storage1";
+
+    let status_router = service_status_router(&dbus, DBUS_SERVICE, DBUS_PATH).await?;
+    let progress_router = progress_router(&dbus, DBUS_SERVICE, DBUS_PATH).await?;
+    let issues_router = issues_router(&dbus, DBUS_SERVICE, DBUS_PATH).await?;
+
+    let client = StorageClient::new(dbus.clone()).await?;
+    let state = StorageState { client };
+    let router = Router::new()
+        .route("/devices/dirty", get(devices_dirty))
```

Comment:
I wonder whether it might be useful to have a more generic endpoint which could have more "status-like" attributes (apart from `dirty`). But if there is no need...

Reply:
well, more generic will be probably that proposal settings. Currently for devices I am aware only about this dirty flag that signal that re probe is needed. ( and staging can be no longer valid )